src/test/unit/controls/Graph/GraphAxes-spec.js

```diff
@@ -408,6 +408,47 @@ describe("Graph - Axes", () => {
             const rangeY2 = 250;
             expect(getAverageTicksCount(rangeY, rangeY2)).toEqual(5);
         });
+        it("Suppresses tick values's trailing zeros when default d3 tick formatting is used and suppressTrailingZeros is set to true", () => {
```

Comment:
Can we add below testing considerations as well if not converted in the existing test cases? and also Are these cover X and Y and Y-2 axis well?  

- when default d3 tick formatting is **not used** `suppressTrailingZeros` is set to **true**.

- when default d3 tick formatting is **used** suppressTrailingZeros is set to **false**

- when default d3 tick formatting is **not used** `suppressTrailingZeros` is set to **false**

Probably, we may add a test case if x-axis type is default and not default based on the changes added

```js
            1. describe("when default d3 tick formatting is not used suppressTrailingZeros is set to true", () => {
                it("should suppresses tick values's trailing zeros ", () => {
                ...................................................................................

            2. describe("when default d3 tick formatting is used suppressTrailingZeros is set to false", () => {
                it("should not suppresses tick values's trailing zeros ", () => {
                ...................................................................................

            3. describe("when default d3 tick formatting is not used suppressTrailingZeros is set to false", () => {
                it("should not suppresses tick values's trailing zeros ", () => {
                ...................................................................................
```

Reply:
I have added unit test cases for the scenarios [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/1914cd49bbb91bc2b5b66782c74be5ba5475f5c1). Also a combination of `suppressTrailingZeros` (either set to true/false) & default d3 tick formatting is **not used** will not arise. Because `suppressTrailingZeros` is only to trim the trailing zeros when default d3 tick formatting is used.

I have added test cases for the following scenarios:
When default d3 tick formatting is used and suppressTrailingZeros is set to true
When default d3 tick formatting is used and suppressTrailingZeros is set to false
When default d3 tick formatting is not used and consumer specifies ~ in tick format
When default d3 tick formatting is not used and consumer does not specify ~ in tick format

I have added unit test cases for all the above 4 scenarios for x, y and y2-axes as well. I have covered the scenario when x-axis type is default. The other scenario which is x-axis type is TIME_SERIES need not be taken into account here since the values are going to be time based, so we need not worry about trailing zeros in this case.